### ISSUE TITLE
Naming strategy UnderScoreSeparatedLowerCase does not work with two Upper characters after each other

### ISSUE THREAD

**Issue by Michal-Danihelka** [micronaut-projects/micronaut-core#4389]
### Task List

- [x] Steps to reproduce provided
- [] Stacktrace (if present) provided
- [] Example that reproduces the problem uploaded to Github
- [x] Full description of the issue provided (see below)

### Steps to Reproduce

1. Call io.micronaut.core.naming.NameUtils.separateCamelCase("DataCJpa", false, '_')

### Expected Behaviour

should return 'Data_C_Jpa' 

### Actual Behaviour

returns 'Data_CJpa'

### Environment Information

- **Operating System**: Windows 10
- **Micronaut Version:** 2.1.0
- **JDK Version:** 11.0.8

### Example Application

N/A

### How to fix

In class io.micronaut.core.naming.NameUtils in method separateCamelCase on line 370:

change line:
if (Character.isUpperCase(c) && !Character.isUpperCase(last)) { 

to:
if (Character.isUpperCase(c)) {

**Comment by jameskleeh** [comment]
That method is private. Under what public method are you calling it?

**Comment by Michal-Danihelka** [comment]
I copied method body to separate class to test it - it is possible to test it independently

**Comment by jameskleeh** [comment]
@Michal-Danihelka You would have to describe the issue as it pertains to one of the public methods in the class. We shouldn't change the method without a just cause that pertains to its designed use cases.

**Comment by Michal-Danihelka** [comment]
1. The main problem is that i have this class:
import io.micronaut.core.annotation.Introspected;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;
import lombok.Data;

import java.util.Date;

//@MappedEntity(value="mpsv_c_NamingStrategywebova_sluzba", escape = false)
@MappedEntity(escape = false)
@Introspected
@Data
public class MpsvCWebovaSluzba {

    @Id
    private int kod;
    private String text;
    private Long poradi;
    private Date zmenaMpsv;


}

2. And this repository - by the way is somewhere described how exactly DSL work in micronaut for micronaut data jdbc?
import cz.oksystem.cks.model.MpsvCWebovaSluzba;
import io.micronaut.data.jdbc.annotation.JdbcRepository;
import io.micronaut.data.repository.CrudRepository;


@JdbcRepository
public interface MpsvCWebovaSluzbaRepository extends CrudRepository<MpsvCWebovaSluzba, Integer> {
    String findTextByKod(int kod);
}

3. Test is:
@MicronautTest
public class MpsvCWebovaSluzbaRepositoryTest {

    @Inject
    MpsvCWebovaSluzbaRepository repository;

    @Test
    void findText() {
        String text = repository.findTextByKod(10);
    }
}

4.  Wrong table name is seen in log:
Caused by: Error : 942, Position : 38, Sql = SELECT mpsv_cwebova_sluzba_.text FROM mpsv_cwebova_sluzba mpsv_cwebova_sluzba_ WHERE (mpsv_cwebova_sluzba_.kod = :1 ), OriginalSql = SELECT mpsv_cwebova_sluzba_.text FROM mpsv_cwebova_sluzba mpsv_cwebova_sluzba_ WHERE (mpsv_cwebova_sluzba_.kod = ?), Error Msg = ORA-00942: table or view does not exist

**Comment by jameskleeh** [comment]
@Michal-Danihelka I don't think we can change this behavior. Doing so would be a breaking change

**Comment by Michal-Danihelka** [comment]
Ok, we will write own NamingStrategy because writing something like this
@MappedEntity(value="mpsv_c_webova_sluzba", escape = false)
is ugly. 
Or do we have other options?

**Comment by jameskleeh** [comment]
I'm not sure offhand. I'll transfer the issue to micronaut-data so it will have visibility there
